### ISSUE TITLE
dlv panics when running dlv config from a dap session

### ISSUE THREAD

**Issue by alexsaezm** [go-delve/delve#3343]
1. What version of Delve are you using (`dlv version`)?
`1.20.1`

2. What version of Go are you using? (`go version`)?
`go version go1.20.3`

3. What operating system and processor architecture are you using?
`linux/amd64`

4. What did you do?
Open VSCode, start a debugging session and in the debug console typed `dlv config`

5. What did you expect to see?
Not a panic.

6. What did you see instead?

```
2023-04-25T20:02:04+02:00 error layer=dap recovered panic: reflect: reflect.Value.Set using value obtained using unexported field
goroutine 6 [running]:
runtime/debug.Stack()
	/usr/lib/golang/src/runtime/debug/stack.go:24 +0x65
github.com/go-delve/delve/service/dap.(*Session).recoverPanic(0xc0003c8000, {0xb86bc0, 0xc0009898c0})
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/server.go:557 +0x66
panic({0x9cf500, 0xc0001c10f0})
	/usr/lib/golang/src/runtime/panic.go:890 +0x263
reflect.flag.mustBeAssignableSlow(0x9cf500?)
	/usr/lib/golang/src/reflect/value.go:259 +0xc5
reflect.flag.mustBeAssignable(...)
	/usr/lib/golang/src/reflect/value.go:249
reflect.Value.Set({0x9c8800?, 0xc0003c8020?, 0x1?}, {0x9c8800?, 0xc0009c1600?, 0x7a4f78?})
	/usr/lib/golang/src/reflect/value.go:2230 +0x6a
github.com/go-delve/delve/pkg/config.ConfigureSetSimple({0x0, 0x0}, {0x0, 0x0}, {0x9c8800?, 0xc0003c8020?, 0x1?})
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/pkg/config/split.go:104 +0x1b8
github.com/go-delve/delve/service/dap.configureSet(0x0?, {0x0?, 0xc0009c15f0?})
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/config.go:39 +0x172
github.com/go-delve/delve/service/dap.(*Session).evaluateConfig(0xc0003c8000, 0x6?, 0x1?, {0x0, 0x0})
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/command.go:123 +0x10e
github.com/go-delve/delve/service/dap.(*Session).delveCmd(0xc0003c8000, 0xa?, 0xaa3025?, {0xc0009c15f0?, 0x0?})
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/command.go:24 +0x404
github.com/go-delve/delve/service/dap.(*Session).onEvaluateRequest(0xc0003c8000, 0xc0009898c0)
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/server.go:2637 +0x9cc
github.com/go-delve/delve/service/dap.(*Session).handleRequest(0xc0003c8000, {0xb86bc0?, 0xc0009898c0?})
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/server.go:741 +0x1251
github.com/go-delve/delve/service/dap.(*Session).ServeDAPCodec(0xc0003c8000)
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/server.go:543 +0x385
github.com/go-delve/delve/service/dap.(*Server).runSession(0xc0001c5680, {0x7f19b8069fc8?, 0xc000386000?})
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/server.go:480 +0xd3
github.com/go-delve/delve/service/dap.(*Server).Run.func1()
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/server.go:472 +0x108
created by github.com/go-delve/delve/service/dap.(*Server).Run
	/home/alex/Developer/pkg/mod/github.com/go-delve/delve@v1.20.1/service/dap/server.go:454 +0x65
```

**Comment by aarzilli** [comment]
Good thing #3335 hadn't been merged yet or it would have looked like my fault.

**Comment by aarzilli** [comment]
Fixed by 13ad7dc

**Comment by nyngwang** [comment]
This still happens to me.
